src/type_methods.jl

```diff
@@ -0,0 +1,610 @@
+# defines common methods for the various rotation types
```

Comment:
The code gen methods have been cut back to just:
vector import / export
basic maths
conversion pathways (e.g. euler -> SpQuat)